src/main/kotlin/com/ajaxproject/warehouse/repository/MongoProductRepositoryImpl.kt

```diff
@@ -0,0 +1,36 @@
+package com.ajaxproject.warehouse.repository
+
+import com.ajaxproject.warehouse.entity.MongoProduct
+import org.bson.types.ObjectId
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.findAll
+import org.springframework.data.mongodb.core.findAndRemove
+import org.springframework.data.mongodb.core.findById
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.data.mongodb.core.query.Query
+import org.springframework.stereotype.Repository
+
+@Repository
+class MongoProductRepositoryImpl(
+    val mongoTemplate: MongoTemplate
+) : MongoProductRepository {
+
+    override fun findAll(): List<MongoProduct> = mongoTemplate.findAll<MongoProduct>(MongoProduct.COLLECTION_NAME)
+
+    override fun getById(id: ObjectId): MongoProduct? =
```

Comment:
```suggestion
    override fun findById(id: ObjectId): MongoProduct? =
```
here you return result or null, this is `find` approach. fix in all similar places. `get` approach is result or error